src/core/sys/linux/fs.d

```diff
@@ -0,0 +1,274 @@
+    c_ulong max_files; /* tunable */
+}
+
+struct inodes_stat_t
+{
+    c_long nr_inodes;
+    c_long nr_unused;
+    c_long[5] dummy; /* padding for sysctl ABI compatibility */
+}
+
+enum NR_FILE = 8192; /* this can well be larger on a larger system */
+
+/*
+ * Structure for FS_IOC_FSGETXATTR[A] and FS_IOC_FSSETXATTR.
+ */
+struct fsxattr
+{
+    uint fsx_xflags; /* xflags field value (get/set) */
+    uint fsx_extsize; /* extsize field value (get/set) */
+    uint fsx_nextents; /* nextents field value (get) */
+    uint fsx_projid; /* project identifier (get/set) */
+    uint fsx_cowextsize; /* CoW extsize field value (get/set) */
+    ubyte[8] fsx_pad;
+}
+
+/*
+ * Flags for the fsx_xflags field
+ */
+enum FS_XFLAG_REALTIME = 0x00000001; /* data in realtime volume */
+enum FS_XFLAG_PREALLOC = 0x00000002; /* preallocated file extents */
+enum FS_XFLAG_IMMUTABLE = 0x00000008; /* file cannot be modified */
+enum FS_XFLAG_APPEND = 0x00000010; /* all writes append */
+enum FS_XFLAG_SYNC = 0x00000020; /* all writes synchronous */
+enum FS_XFLAG_NOATIME = 0x00000040; /* do not update access time */
+enum FS_XFLAG_NODUMP = 0x00000080; /* do not include in backups */
+enum FS_XFLAG_RTINHERIT = 0x00000100; /* create with rt bit set */
+enum FS_XFLAG_PROJINHERIT = 0x00000200; /* create with parents projid */
+enum FS_XFLAG_NOSYMLINKS = 0x00000400; /* disallow symlink creation */
+enum FS_XFLAG_EXTSIZE = 0x00000800; /* extent size allocator hint */
+enum FS_XFLAG_EXTSZINHERIT = 0x00001000; /* inherit inode extent size */
+enum FS_XFLAG_NODEFRAG = 0x00002000; /* do not defragment */
+enum FS_XFLAG_FILESTREAM = 0x00004000; /* use filestream allocator */
+enum FS_XFLAG_DAX = 0x00008000; /* use DAX for IO */
+enum FS_XFLAG_COWEXTSIZE = 0x00010000; /* CoW extent size allocator hint */
+enum FS_XFLAG_HASATTR = 0x80000000; /* no DIFLAG for this */
+
+/* the read-only stuff doesn't really belong here, but any other place is
+   probably as bad and I don't want to create yet another include file. */
+
+enum BLKROSET = _IO(0x12, 93); /* set device read-only (0 = read-write) */
+enum BLKROGET = _IO(0x12, 94); /* get read-only status (0 = read_write) */
+enum BLKRRPART = _IO(0x12, 95); /* re-read partition table */
+enum BLKGETSIZE = _IO(0x12, 96); /* return device size /512 (long *arg) */
+enum BLKFLSBUF = _IO(0x12, 97); /* flush buffer cache */
+enum BLKRASET = _IO(0x12, 98); /* set read ahead for block device */
+enum BLKRAGET = _IO(0x12, 99); /* get current read ahead setting */
+enum BLKFRASET = _IO(0x12, 100); /* set filesystem (mm/filemap.c) read-ahead */
+enum BLKFRAGET = _IO(0x12, 101); /* get filesystem (mm/filemap.c) read-ahead */
+enum BLKSECTSET = _IO(0x12, 102); /* set max sectors per request (ll_rw_blk.c) */
+enum BLKSECTGET = _IO(0x12, 103); /* get max sectors per request (ll_rw_blk.c) */
+enum BLKSSZGET = _IO(0x12, 104); /* get block device sector size */
+
+/* See blkpg.h */
+
+/* Some people are morons.  Do not use sizeof! */
```

Comment:
Do we really need all comments included ad verbatim? Just ddoc `///` markers will be enough.

Reply:
Yeah, I'm going to remove some of those from here and transform what should be useful for documentation to Ddoc comments.

Reply:
Done